### ISSUE TITLE
Enable the selection when multiple layers are shown

### ISSUE THREAD

**Issue by PennyQ** [glue-viz/glue-vispy-viewers#93]
Combine multiple 'visible' layer data when selecting to make sure 'what you select is what you see'.

For scatter points it could be simply merge all 'visible' layer data into one large array, and do the selection on this array. But the volume rendering seems more complex.

**Comment by PennyQ** [comment]
@astrofrog I tried to get a mask of combined visible_data for scatter points, but I found this `mark_selected` function doesn't properly support `visible_data` with len>1 and seems the length of `mask` of should be identical with the subset. 

Can we improve this? (listed with the code below)

```
def mark_selected(self, mask, visible_data):
        subset_state = ElementSubsetState(np.where(mask)[0])
        mode = EditSubsetMode()
        focus = visible_data[0] if len(visible_data) > 0 else None
        mode.update(self._data_collection, subset_state, focus_data=focus)
```

**Comment by astrofrog** [comment]
@PennyQ - just FYI, I fixed this in #208
